### DISCUSSION TITLE
ATTiny88 no longer works after Micronucleus push

### DISCUSSION THREAD

**Discussion by Neciota** [SpenceKonde/ATTinyCore#708] [Support / Q & A]
I'm very new to the Tiny and microcontroller programming in general. I was pushing my code in the Arduino IDE to my ATTinyCore 88 via Micronucleus, but it lost connection during the upload (somehow) and afterwards it was no longer recognized by my computer. 

Device manager indicates:
![image](https://user-images.githubusercontent.com/18664259/181349140-6618603e-3b77-42db-8b96-2281ae6883df.png)

If I try to push now it simply won't find the device. Unfortunately I did not save the error log from the IDE after the issue occurred, and obviously now I cannot reproduce it. The device is depicted as above on both my laptop and desktop, so I assume it is an issue with the ATTiny. Could I "factory reset" it via USB? I don't think I have the materials to program via ISP.

**Comment by pcfreak1201** [comment]
The Micronucleus stuff is software USB (V-USB), so if it is "bricked", you have to use ISP to burn the bootloader again:
https://github.com/SpenceKonde/ATTinyCore/blob/master/avr/extras/UsingMicronucleus.md#installing-micronucleus-via-isp

**Reply by Neciota** [reply]
When using ISP I see a lot of examples online of programming the individual controller, but I got a Nano V3.0 with it on there. Do I just connect the pins from an Uno to my Nano's board to program it, or do I need the additional capacitors and resistor?

**Comment by pcfreak1201** [comment]
Well, I think you have this one: https://github.com/SpenceKonde/ATTinyCore/issues/435 ?
If yes, you have an ISP-Connector, where you can connect any ISP-Programmer or UNO as ISP. For the UNO, you need the 10uF capacitor (see Arduino as ISP), plus (Dupont-) wires...
Select your programmer, interface (COM ...) and Burn Burnloader: Fresh Install via ISP; press Burn Bootloader to go

**Comment by prandeamus** [comment]
If I remember correctly, micronucleus only presents its USB interface for a short time. On the digispark cards that use ATTiny85, they usually ship with micronucleus "listening" for 30 seconds or so. After that time, the bootloader tries to run the code. Nothing is then polling the USB hardware and it appears to fail from the windows client. One possible "reason" for the fault here is that the timeout has passed before you tried to download. Unlike the classic Nano boot loader, you can't just press a button in Arduino and off you go. Micronucleus as shipped will require a reset or power cycle. May not apply in your case, but worth considering.
